backend/cmd/user/import.go

```diff
@@ -0,0 +1,197 @@
+package user
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gobuffalo/pop/v6"
+	"github.com/gofrs/uuid"
+	"github.com/spf13/cobra"
+	"github.com/teamhanko/hanko/backend/config"
+	"github.com/teamhanko/hanko/backend/persistence"
+	"github.com/teamhanko/hanko/backend/persistence/models"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+func NewImportCommand() *cobra.Command {
+	var (
+		configFile string
+		inputFile  string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "Import users into database from a Json file",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			//Load cfg
+			cfg, err := config.Load(&configFile)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			//Load File
+			// we explicitly want user input here, hence  #nosec G304
+			jsonFile, err := os.Open(inputFile)
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer func() {
+				if err := jsonFile.Close(); err != nil {
+					log.Printf("Error closing file: %s\n", err)
+				}
+			}()
+			users, err := loadAndValidate(jsonFile)
+			if err != nil {
+				log.Fatal(err)
+			}
+			//Validate Input
+			err = validateEntries(users)
+			if err != nil {
+				log.Fatal(err)
+			}
+			//Import Users
+			persister, err := persistence.New(cfg.Database)
+			if err != nil {
+				log.Fatal(err)
+			}
+			err = addToDatabase(users, persister)
+			if err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&configFile, "config", config.DefaultConfigFilePath, "config file")
+	cmd.Flags().StringVarP(&inputFile, "inputFile", "i", "", "The json file where the users should be imported from.")
+	err := cmd.MarkFlagRequired("inputFile")
+	if err != nil {
+		log.Println(err)
+	}
+	return cmd
+}
+
+// loadAndValidate reads json from an io.Reader so we read every entry separate and validate it. We go through the whole
+// array to print out every validation error in the input data.
+func loadAndValidate(input io.Reader) ([]ImportEntry, error) {
+	dec := json.NewDecoder(input)
+
+	// read the open bracket
+	_, err := dec.Token()
+	if err != nil {
+		return nil, err
+	}
+
+	users := []ImportEntry{}
+
+	numErrors := 0
+	index := 0
+	// while the array contains values
+	for dec.More() {
+		index = index + 1
+		var userEntry ImportEntry
+		// decode one ImportEntry
+		err := dec.Decode(&userEntry)
+		if err != nil {
+			errorMsg := fmt.Sprintf("Error at entry %v : %v", index, err.Error())
+			log.Println(errorMsg)
+			return nil, err
+		}
+
+		if err := userEntry.validate(); err != nil {
+			errorMsg := fmt.Sprintf("Error at entry %v : %v", index, err.Error())
+			log.Println(errorMsg)
+			log.Print(userEntry)
+			numErrors++
+			continue
+		}
+		users = append(users, userEntry)
+	}
+
+	// read closing bracket
+	_, err = dec.Token()
+	if err != nil {
+		return nil, err
+	}
+	if numErrors > 0 {
+		errMsg := fmt.Sprintf("Found %v errors.", numErrors)
+		log.Printf(errMsg)
```

Comment:
This log is redundant, because the error from this function (which is this message) is also returned from `log.Fatal` from the calling function.